cmd/run.go

```diff
@@ -383,6 +352,144 @@ func Run(isDebug *bool) *cli.Command {
 	}
 }
 
+func ReadState(fs afero.Fs, statePath string, filter *Filter) (*scheduler.PipelineState, error) {
+	pipelineState, err := scheduler.ReadState(fs, statePath)
+	if err != nil {
+		errorPrinter.Printf("Failed to restore state: %v\n", err)
+		return nil, err
+	}
+
+	filter.IncludeTag = pipelineState.Parameters.Tag
+
+	filter.OnlyTaskTypes = pipelineState.Parameters.Only
+
+	filter.PushMetaData = pipelineState.Parameters.PushMetadata
+
+	filter.ExcludeTag = pipelineState.Parameters.ExcludeTag
+
+	return pipelineState, nil
+}
+
+func GetPipeline(inputPath string, runConfig *scheduler.RunConfig, logger *zap.SugaredLogger) (*pipeline.Pipeline, bool, bool, *config.Config, error) {
+	pipelinePath := inputPath
+	repoRoot, err := git.FindRepoFromPath(inputPath)
+	if err != nil {
+		errorPrinter.Printf("Failed to find the git repository root: %v\n", err)
+		return nil, false, false, nil, err
+	}
+
+	runningForAnAsset := isPathReferencingAsset(inputPath)
+	if runningForAnAsset && runConfig.Tag != "" {
+		errorPrinter.Printf("You cannot use the '--tag' flag when running a single asset.\n")
+		return nil, false, false, nil, err
+	}
+
+	var task *pipeline.Asset
+	runDownstreamTasks := false
+	if runningForAnAsset {
+		task, err = DefaultPipelineBuilder.CreateAssetFromFile(inputPath)
+		if err != nil {
+			errorPrinter.Printf("Failed to build asset: %v. Are you sure you used the correct path?\n", err.Error())
+
+			return nil, runDownstreamTasks, runningForAnAsset, nil, err
+		}
+		if task == nil {
+			errorPrinter.Printf("The given file path doesn't seem to be a Bruin task definition: '%s'\n", inputPath)
+			return nil, runDownstreamTasks, runningForAnAsset, nil, err
+		}
+
+		pipelinePath, err = path.GetPipelineRootFromTask(inputPath, pipelineDefinitionFiles)
+		if err != nil {
+			errorPrinter.Printf("Failed to find the pipeline this task belongs to: '%s'\n", inputPath)
+			return nil, runDownstreamTasks, runningForAnAsset, nil, err
+		}
+
+		if runConfig.Downstream {
+			infoPrinter.Println("The downstream tasks will be executed as well.")
+			runDownstreamTasks = true
+		}
```

Comment:
this logic could go outside of this function it seems like, since it is not very relevant to what's being done here and simply being returned.